qiskit/pulse/schedule.py

```diff
@@ -926,65 +478,1109 @@ def draw(self,
+    over all input signal *channels* and supporting special syntaxes for building.
 
-        # then verify same number of instructions in each
-        instructions = self.instructions
-        other_instructions = other.instructions
-        if len(instructions) != len(other_instructions):
-            return False
+    In Qiskit Pulse model [1], time overlap between pulse instructions on the same channel
+    is not allowed. This overlap constraint is immediately evaluated when new
+    instruction is added to the ``Schedule`` object.
 
-        # finally check each instruction in `other` is in this schedule
-        for idx, inst in enumerate(other_instructions):
-            # check assumes `Schedule.instructions` is sorted consistently
-            if instructions[idx] != inst:
-                return False
+    This indicates we always need to specify absolute values for instruction start time ``t0``
+    and duration of the instruction to calculate ``timeslots`` occupation by the instruction.
 
-        return True
+    The ``Schedule`` program supports some syntax sugar for efficient programming.
 
-    def __add__(self, other: ScheduleComponent) -> 'Schedule':
-        """Return a new schedule with ``other`` inserted within ``self`` at ``start_time``."""
-        return self.append(other)
+    - Appending instruction at the end of channel
 
-    def __or__(self, other: ScheduleComponent) -> 'Schedule':
-        """Return a new schedule which is the union of `self` and `other`."""
-        return self.insert(0, other)
+    .. code-block:: python
 
-    def __lshift__(self, time: int) -> 'Schedule':
-        """Return a new schedule which is shifted forward by ``time``."""
-        return self.shift(time)
+        sched = Schedule()
+        sched += Play(Gaussian(160, 0.1, 40), DriveChannel(0))
 
-    def __len__(self) -> int:
-        """Return number of instructions in the schedule."""
-        return len(self.instructions)
+    - Appending instruction with time shift
+
+    .. code-block:: python
+
+        sched = Schedule()
+        sched += Play(Gaussian(160, 0.1, 40), DriveChannel(0)) << 30
+
+    - Merging two schedule
+
+    .. code-block:: python
+
+        sched1 = Schedule()
+        sched1 += Play(Gaussian(160, 0.1, 40), DriveChannel(0))
+
+        sched2 = Schedule()
+        sched2 += Play(Gaussian(160, 0.1, 40), DriveChannel(1))
+        sched2 = sched1 | sched2
+
+    The ``PulseError`` is immediately raised when the overlap constraint is violated.
+
+    In the schedule representation, we cannot parametrize duration of waveforms.
+    Thus we need to create new schedule object for every duration, i.e. Rabi experiment.
+
+    References:
+        [1]: https://arxiv.org/abs/2004.06755
+
+    """
+    # Prefix to use for auto naming.
+    prefix = 'sched'
+
+    # Counter to count instance number.
+    instances_counter = itertools.count()
```

Comment:
To avoid `ScheduleBlock` and `Schedule` to share the index. 
```
block0 = ScheduleBlock() # block-0
sched0 = Schedule()  # sched-1
block1 = ScheduleBlock() # block-2
```

Reply:
ok